File: src/app/box/test1/CsvToJsonConverter.java
import java.io.*;
import java.nio.charset.StandardCharsets;
import java.util.*;

public class CsvToJsonConverter {

    public static void main(String[] args) {
        String inputDir = ".";
        String outputDir = ".";

        // CSVファイルをJSONに変換
        convertCsvToJson(inputDir + "/file_categories.csv", outputDir + "/file_categories.json");
        convertCsvToJson(inputDir + "/file_details.csv", outputDir + "/file_details.json");
        convertCsvToJson(inputDir + "/file_technologies.csv", outputDir + "/file_technologies.json");

        System.out.println("CSVからJSONへの変換が完了しました。");
    }

    public static void convertCsvToJson(String csvFilePath, String jsonFilePath) {
        List<Map<String, String>> records = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(csvFilePath), StandardCharsets.UTF_8))) {
            String line;
            String[] headers = null;
            boolean isFirstLine = true;

            while ((line = br.readLine()) != null) {
                // タブ区切りで分割
                String[] values = line.split("\t");

                if (isFirstLine) {
                    // ヘッダー行
                    headers = values;
                    isFirstLine = false;
                } else {
                    // データ行
                    Map<String, String> record = new LinkedHashMap<>();
                    for (int i = 0; i < headers.length && i < values.length; i++) {
                        String value = values[i];

                        // BOX IDの場合、シングルクォートを除去
                        if (headers[i].contains("BOX ID") || headers[i].equals("ファイル BOX ID")) {
                            value = value.replaceAll("'", "");
                        }

                        record.put(headers[i], value);
                    }
                    records.add(record);
                }
            }
        } catch (IOException e) {
            System.err.println("CSVファイルの読み込みエラー: " + csvFilePath);
            e.printStackTrace();
            return;
        }

        // JSONファイルに書き込み
        try (PrintWriter writer = new PrintWriter(new OutputStreamWriter(new FileOutputStream(jsonFilePath), StandardCharsets.UTF_8))) {
            writer.println("[");
            for (int i = 0; i < records.size(); i++) {
                Map<String, String> record = records.get(i);
                writer.println("  {");

                int fieldCount = 0;
                for (Map.Entry<String, String> entry : record.entrySet()) {
                    writer.print("    \"" + escapeJsonString(entry.getKey()) + "\": \"" + escapeJsonString(entry.getValue()) + "\"");
                    fieldCount++;
                    if (fieldCount < record.size()) {
                        writer.println(",");
                    } else {
                        writer.println();
                    }
                }

                writer.print("  }");
                if (i < records.size() - 1) {
                    writer.println(",");
                } else {
                    writer.println();
                }
            }
            writer.println("]");

            System.out.println(jsonFilePath + " に変換完了");
        } catch (IOException e) {
            System.err.println("JSONファイルの書き込みエラー: " + jsonFilePath);
            e.printStackTrace();
        }
    }

    private static String escapeJsonString(String str) {
        if (str == null) return "";
        return str.replace("\\", "\\\\")
                  .replace("\"", "\\\"")
                  .replace("\n", "\\n")
                  .replace("\r", "\\r")
                  .replace("\t", "\\t");
    }
}

File: src/app/box/test1/csvToJsonConverter.js
const fs = require('fs');
const path = require('path');

function csvToJson(csvFilePath, jsonFilePath) {
    try {
        // CSVファイルを読み込み
        const csvData = fs.readFileSync(csvFilePath, 'utf8');

        // Windowsの改行コードを統一
        const normalizedData = csvData.replace(/\r\n/g, '\n').replace(/\r/g, '\n');

        // 空行を除去して行を分割
        const lines = normalizedData.split('\n').filter(line => line.trim().length > 0);

        console.log(`CSVファイル ${csvFilePath} の総行数: ${lines.length}`);

        if (lines.length === 0) {
            console.error('CSVファイルが空です');
            return;
        }

        // ヘッダー行を処理（タブ区切り）
        const headers = lines[0].split('\t').map(header => header.trim());
        console.log(`ヘッダー数: ${headers.length}`, headers);

        const records = [];

        // データ行を処理
        for (let i = 1; i < lines.length; i++) {
            const line = lines[i].trim();
            if (!line) continue;

            const values = line.split('\t');
            console.log(`行 ${i}: 値の数 ${values.length}`);

            const record = {};

            for (let j = 0; j < headers.length && j < values.length; j++) {
                let value = values[j].trim();

                // BOX IDの場合、シングルクォートを除去
                if (headers[j].includes('BOX ID') || headers[j] === 'ファイル BOX ID') {
                    value = value.replace(/'/g, '');
                }

                // 値がnullや空文字の場合はnullに統一
                if (value === 'null' || value === '') {
                    value = null;
                }

                record[headers[j]] = value;
            }

            records.push(record);
        }

        console.log(`変換されたレコード数: ${records.length}`);
        console.log('最初のレコードのサンプル:', records[0]);

        // JSONファイルに書き込み
        fs.writeFileSync(jsonFilePath, JSON.stringify(records, null, 2), 'utf8');
        console.log(`${jsonFilePath} に変換完了 (${records.length} 件のデータ)`);

    } catch (error) {
        console.error(`エラー: ${csvFilePath} の変換に失敗しました`, error);
    }
}

function main() {
    const inputDir = __dirname;

    // 3つのCSVファイルをJSONに変換
    csvToJson(
        path.join(inputDir, 'file_categories.csv'),
        path.join(inputDir, 'file_categories.json')
    );

    csvToJson(
        path.join(inputDir, 'file_details.csv'),
        path.join(inputDir, 'file_details.json')
    );

    csvToJson(
        path.join(inputDir, 'file_technologies.csv'),
        path.join(inputDir, 'file_technologies.json')
    );

    console.log('全てのCSVからJSONへの変換が完了しました。');
}

main();
